sale_loyalty_auto_refresh/models/sale_order.py

```diff
@@ -40,10 +44,23 @@ def write(self, vals):
         return res
 
     def _auto_refresh_coupons(self):
-        orders = self.filtered(type(self)._allow_recompute_coupon_lines)
-        if orders:
-            orders = orders.with_context(skip_auto_refresh_coupons=True)
-            orders.recompute_coupon_lines()
+        orders = self.with_context(skip_auto_refresh_coupons=True).filtered(
+            type(self)._allow_recompute_coupon_lines
+        )
+        for order in orders:
+            order._update_programs_and_rewards()
+            order.action_apply_rewards()
+
+    def action_apply_rewards(self):
+        self.ensure_one()
+        claimable_rewards = self._get_claimable_rewards()
+        for coupon, reward in claimable_rewards.items():
+            try:
+                self._apply_program_reward(reward, coupon)
+                self._update_programs_and_rewards()
+            except Exception as e:
```

Comment:
Isn't it a too broad except ?

Reply:
yes, but here, i want to ignore all exception errors to unblock the user when creating/writing

Reply:
anw, i updated to catch UserError, ValidationError only.